x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/util/DateUtils.java

```diff
@@ -17,38 +17,82 @@
 import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
 
-public class DateUtils {
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+
+public final class DateUtils {
+
+    private static final long DAY_IN_MILLIS = 60 * 60 * 24 * 1000;
 
     // TODO: do we have a java.time based parser we can use instead?
     private static final DateTimeFormatter UTC_DATE_FORMATTER = ISODateTimeFormat.dateOptionalTimeParser().withZoneUTC();
 
-    public static ZoneId UTC = ZoneId.of("Z");
+    public static final ZoneId UTC = ZoneId.of("Z");
 
     private DateUtils() {}
 
+    /**
+     * Creates an date for SQL DATE type from the millis since epoch.
+     */
+    public static ZonedDateTime asDateOnly(long millis) {
+        return ZonedDateTime.ofInstant(Instant.ofEpochMilli(millis), UTC)
```

Comment:
The `with` methods are useful when changing only one param - here however 3 intermediate objects are created.
Use `ZonedDateTime.truncatedTo` instead (recommended) or `ZonedDateTime.of`